### PR TITLE
ENH: Add expression language parser

### PR DESCRIPTION
**Pull request by effigies**
Adding a Python parser for the schema expression language

### CODE REVIEW

File: tools/schemacode/bidsschematools/expressions.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+# Arithmetic expressions
+factor <<= atom + (expOp + factor)[...]  # Right-associative
+term = factor + (mulOp + factor)[...]
+expr = term + (addOp + term)[...]
+
+# Logic expressions (tests, to avoid name collision)
+comparison = expr + (compOp + expr)[...]
+notTest <<= notOp + notTest | comparison
+andTest = notTest + (andOp + notTest)[...]
```
Comment (effigies) [inline tools/schemacode/bidsschematools/expressions.py:99]: Should be reverted to right associative to simplify short circuiting.

File: tools/schemacode/bidsschematools/expressions.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+test <<= andTest + (orOp + andTest)[...]
```
Comment (effigies) [inline tools/schemacode/bidsschematools/expressions.py:100]: And here

**Comment by effigies** [comment]
@rwblair AST complete. What do you think? Should I go ahead and implement interpretation before merging, or get this in and make interpretation a new PR? And any other tests you can think of?

**Comment by rwblair** [comment]
Looks great. For a real thorough test could try and parse every selector and check.

Guess go ahead an merge lest it turn into one of my validator PRs...

**Comment by effigies** [comment]
Tests added. Which goes to show that parsing isn't type checking... Either way, let's get moving.

**Pull request merged**